src/Analyzers/CSharp/Tests/UseIndexOrRangeOperator/UseRangeOperatorTests.cs

```diff
@@ -479,5 +478,34 @@ void Goo(Span<int> s)
                 FixedCode = fixedSource,
             }.RunAsync();
         }
+
+        [WorkItem(43202, "https://github.com/dotnet/roslyn/issues/43202")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseRangeOperator)]
+        public async Task TestWritableType()
+        {
+            var source =
+@"
+using System;
+struct S { 
+    public ref S Slice(int start, int length) => throw null; 
+    public int Length { get; } 
+    public S this[System.Range r] { get => default; } 
+}
+
+class C
+{
+    void Goo(S s)
+    {
+        s.Slice(1, s.Length - 2) = default;
+    }
+}";
+
+            await new VerifyCS.Test
+            {
+                ReferenceAssemblies = ReferenceAssemblies.NetCore.NetCoreApp31,
+                TestCode = source,
+                FixedCode = source,
+            }.RunAsync();
```

Comment:
do we have a test where both indexer/method are by-ref?  If not, please add.  THanks!

Reply:
Yes, It's on line 324.
`public async Task TestNonStringType_Assignment()
        {
            var source =
@"
struct S { public ref S Slice(int start, int length) => throw null; public int Length { get; } public ref S this[System.Range r] { get => throw null; } }
class C
{
    void Goo(S s)
    {
        s.Slice([|1, s.Length - 2|]) = default;
    }
}";
            var fixedSource =
@"
struct S { public ref S Slice(int start, int length) => throw null; public int Length { get; } public ref S this[System.Range r] { get => throw null; } }
class C
{
    void Goo(S s)
    {
        s[1..^1] = default;
    }
}";`